src/basicfuns.jl

```diff
@@ -227,46 +227,40 @@ logsubexp(x::Real, y::Real) = max(x, y) + log1mexp(-abs(x - y))
 Compute `log(sum(exp, X))` in a numerically stable way that avoids intermediate over- and
 underflow.
 
-`X` should be an iterator of real numbers.
+`X` should be an iterator of real numbers. The result is computed using a single pass over
+the data.
 
-See also: [`logsumexp_onepass`](@ref)
+# References
+
+[Sebastian Nowozin: Streaming Log-sum-exp Computation.](http://www.nowozin.net/sebastian/blog/streaming-log-sum-exp-computation.html)
 """
 logsumexp(X) = logsumexp_onepass(X)
-logsumexp(X::AbstractArray{<:Real}; dims=:) = _logsumexp(X, dims)
-
-_logsumexp(X::AbstractArray{<:Real}, ::Colon) = logsumexp_onepass(X)
-function _logsumexp(X::AbstractArray{<:Real}, dims)
-    # Do not use log(zero(eltype(X))) directly to avoid issues with ForwardDiff (#82)
-    u = reduce(max, X, dims=dims, init=oftype(log(zero(eltype(X))), -Inf))
-    u isa AbstractArray || isfinite(u) || return float(u)
-    let u=u # avoid https://github.com/JuliaLang/julia/issues/15276
-        # TODO: remove the branch when JuliaLang/julia#31020 is merged.
-        if u isa AbstractArray
-            u .+ log.(sum(exp.(X .- u); dims=dims))
-        else
-            u + log(sum(x -> exp(x-u), X))
-        end
-    end
-end
 
 """
-    logsumexp_onepass(X)
+    logsumexp(X::AbstractArray{<:Real}[; dims=:])
```

Comment:
No need for brackets, they are implicit when there's a default value:
```suggestion
    logsumexp(X::AbstractArray{<:Real}; dims=:)
```